packages/core/src/hydration/event_replay.ts

```diff
@@ -177,73 +183,13 @@ export function setJSActionAttribute(
   }
 }
 
-/**
- * Finds an LView that a given DOM element belongs to.
- */
-function getLViewByElement(target: HTMLElement): LView | null {
-  let lView = readLView(target);
-  if (lView) {
-    return lView;
-  } else {
-    // If this node doesn't have LView info attached, then we need to
-    // traverse upwards up the DOM to find the nearest element that
-    // has already been monkey patched with data.
-    let parent = target as HTMLElement;
-    while ((parent = parent.parentNode as HTMLElement)) {
-      lView = readLView(parent);
-      if (lView) {
-        // To prevent additional lookups, monkey-patch LView id onto this DOM node.
-        // TODO: consider patching all parent nodes that didn't have LView id, so that
-        // we can avoid lookups for more nodes.
-        attachLViewId(target, lView);
-        return lView;
-      }
-    }
-  }
-  return null;
-}
-
 function handleEvent(event: EventInfoWrapper) {
-  const nativeElement = event.getAction()!.element;
-  // Dispatch event via Angular's logic
-  if (nativeElement) {
-    const lView = getLViewByElement(nativeElement as HTMLElement);
-    if (lView !== null) {
-      const tView = lView[TVIEW];
-      const eventName = event.getEventType();
-      const origEvent = event.getEvent();
-      const listeners = getEventListeners(tView, lView, nativeElement, eventName);
-      for (const listener of listeners) {
-        listener(origEvent);
-      }
-    }
+  const nativeElement = unwrapRNode(event.getAction()!.element) as Element;
```

Comment:
We probably don't need the `unwrapRNode` call here as well (since we already have a reference to a DOM node).
```suggestion
  const nativeElement = event.getAction()!.element as Element;
```

Reply:
done, but i needed to cast.